packages/mobile/src/harmony-native/components/input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,311 @@
+import { forwardRef, useCallback, useRef } from 'react'
+
+import { css } from '@emotion/native'
+import type {
+  NativeSyntheticEvent,
+  TextInputFocusEventData
+} from 'react-native'
+import { Pressable, TextInput as RNTextInput } from 'react-native'
+import { Gesture, GestureDetector, State } from 'react-native-gesture-handler'
+import Animated, {
+  interpolate,
+  interpolateColor,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming
+} from 'react-native-reanimated'
+
+import { useTheme } from '@audius/harmony-native'
+
+import { useFocusState } from '../../../hooks/useFocusState'
+import { mergeRefs } from '../../../utils/mergeRefs'
+import type { TextColors, TextSize } from '../../Text/Text'
+import { Text } from '../../Text/Text'
+import { Flex } from '../../layout'
+
+import { TextInputSize, type TextInputProps } from './types'
+
+const AnimatedText = Animated.createAnimatedComponent(Text)
+const AnimatedFlex = Animated.createAnimatedComponent(Flex)
+
+export const TextInput = forwardRef(
+  (props: TextInputProps, ref: Ref<RNTextInput>) => {
+    const {
+      'aria-label': ariaLabel,
+      required,
+      maxLength,
+      showMaxLengthThreshold = 0.7,
+      maxLengthWarningThreshold = 0.9,
+      size = TextInputSize.DEFAULT,
+      hideLabel,
+      label: labelProp,
+      value,
+      id: idProp,
+      warning,
+      error,
+      disabled,
+      onFocus: onFocusProp,
+      onBlur: onBlurProp,
+      placeholder,
+      helperText,
+      startAdornmentText,
+      endAdornmentText,
+      startIcon: StartIcon,
+      endIcon: EndIcon,
+      endAdornment: endAdornmentProp,
+      _incorrectError,
+      _isFocused,
+      _disablePointerEvents,
+      ...other
+    } = props
+
+    const innerInputRef = useRef<RNTextInput>(null)
+
+    const { typography, color, motion } = useTheme()
+
+    let endAdornment: null | ReactNode
+    if (EndIcon != null) {
+      endAdornment = <EndIcon size='m' color='subdued' />
+    } else if (endAdornmentProp != null) {
+      endAdornment = endAdornmentProp
+    } else {
+      endAdornment = null
+    }
+
+    const focused = useSharedValue(_isFocused ? 1 : 0)
+
+    const handleFocus = useCallback(
+      (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        focused.value = withTiming(1, motion.press)
+        onFocusProp?.(e)
+      },
+      [focused, motion.press, onFocusProp]
+    )
+
+    const handleBlur = useCallback(
+      (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        focused.value = withTiming(0, motion.press)
+        onBlurProp?.(e)
+      },
+      [focused, motion.press, onBlurProp]
+    )
+
+    const [isFocusedState, onFocus, onBlur] = useFocusState(
+      handleFocus,
+      handleBlur
+    )
+
+    const isFocused = _isFocused ?? isFocusedState
+
+    // For focus behavior and accessiblity, <label> needs to have a htmlFor={} provided to an id matching the input
+    // const backupId = useId()
```

Comment:
comments 👀

Reply:
ah good find. let me investigate if id's are necessary for input a11y in mobile